source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesAgent/KubernetesAgentMigrateFromPreinstallationTest.cs

```diff
@@ -0,0 +1,189 @@
+using System.Text;
+using FluentAssertions;
+using k8s;
+using k8s.Models;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Commands;
+using Octopus.Tentacle.Diagnostics;
+using Octopus.Tentacle.Startup;
+
+namespace Octopus.Tentacle.Kubernetes.Tests.Integration.KubernetesAgent;
+
+public class KubernetesAgentMigrateFromPreinstallationTest
+{
+    
+    readonly ISystemLog systemLog = new SystemLog();
+    
+    [Test]
+    public async Task MigrationFromPreinstallHookShouldCopyData()
+    {
+        //Arrange
+        var kubernetesConfigClient = new KubernetesFileWrappedProvider(KubernetesTestsGlobalContext.Instance.KubeConfigPath);
+        var commandNamespace = Guid.NewGuid().ToString("N");
+        Environment.SetEnvironmentVariable(KubernetesConfig.NamespaceVariableName, commandNamespace);
+        var commandToRun = new MigratePreInstalledKubernetesDeploymentTargetCommand(new Lazy<IKubernetesClientConfigProvider>(kubernetesConfigClient), systemLog, new LogFileOnlyLogger());
+        var client = new k8s.Kubernetes(kubernetesConfigClient.Get());
+        var validationKey = Guid.NewGuid().ToString("N");
+        var sourceConfigMapData = new Dictionary<string, string>
+        {
+            {"validationKey", validationKey},
+            {"Tentacle.Services.IsRegistered", "true"}
+        };
+        var sourceSecretData = new Dictionary<string, string>
+        {
+            {"machine-iv", "testData"},
```

Comment:
is `iv` a thing?

Reply:
Yes, it's the initialisation vector, used for encrypting log messages.